aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -154,6 +216,54 @@ Tensor & masked_select_out_cuda(Tensor & result, const Tensor & self, const Tens
   return masked_select_out_cuda_impl(result, self, mask);
 }
 
+void take_out_cuda_template(
+  Tensor& output,
+  Tensor const& input,
+  Tensor const& index)
+{
+  TORCH_CHECK(output.device().type() == at::kCUDA, "device type of output (", output.device().type(), ") is not on the GPU");
+  TORCH_CHECK(input.device().type() == at::kCUDA, "device type of input (", input.device().type(), ") is not on the GPU");
+  TORCH_CHECK(index.device().type() == at::kCUDA, "device type of index (", index.device().type(), ") is not on the GPUA");
+
+  TORCH_CHECK(output.layout() == Layout::Strided, "take() only supports strided layout, got layout: ", output.layout(), " on output tensor");
+  TORCH_CHECK(input.layout() == Layout::Strided, "take() only supports strided layout, got layout: ", input.layout(), " on input tensor");
+  TORCH_CHECK(index.layout() == Layout::Strided, "take() only supports strided layout, got layout: ", index.layout(), " on index tensor");
+
+  TORCH_CHECK(output.scalar_type() == input.scalar_type(), "output and input scalar type must match. but got different types: ", output.scalar_type(), " and ", input.scalar_type());
+  TORCH_CHECK(index.scalar_type() == kLong, "index must be an int64 tensor");
+
+  TensorArg output_arg{ output, "output", 1 };
+  TensorArg input_arg{ input, "input", 2 };
+  TensorArg index_arg{ index, "index", 3 };
+  checkAllSameGPU("take", {output_arg, input_arg});
+
+  // TODO: the calls to checkGPU() also assert that all tensors are on CURRENT device, not just the same device as each other (using cudaGetDevice())
+  /*THCAssertSameGPU(THCudaLongTensor_checkGPU(state, 1, index));*/
```

Comment:
See comment- the original code called `THCudaLongTensor_checkGPU`, which checks that the tensor is on the same device as cudaGetDevice(). It doesn't look like the corresponding ATen function `checkAllSameGPU` performs that check, and I couldn't find ATen code that doesn't something similar.

Reply:
I can use that directly. Do you think it would be worth updating `checkAllSameGPU` to do that check too (so that it checks that all passed in tensors are on the same gpu as `current_device()`)? Or would that be too invasive of a change for this PR.

For now I'll just add the extra check directly into the above code.

Reply:
> Do you think it would be worth updating checkAllSameGPU to do that check too (so that it checks that all passed in tensors are on the same gpu as current_device())?

If doing that, please verify that all callsite of `checkAllSameGPU` expect the tensors are on the current device and also please add comments to `checkAllSameGPU` to highlight its behavior..